cms/djangoapps/contentstore/management/commands/swap_from_auto_track_cohort_pilot.py

```diff
@@ -0,0 +1,168 @@
+
+        items = module_store.get_items(course_key)
+        if not items:
+            raise CommandError("Course with %s key not found." % course_id)
+
+        # Get the CourseUserGroup IDs for the audit course names
+        audit_course_user_group_ids = CourseUserGroup.objects.filter(
+            name__in=audit_cohort_names,
+            course_id=course_key,
+            group_type=CourseUserGroup.COHORT,
+        ).values_list('id', flat=True)
+
+        # Get all of the audit CourseCohorts from the above IDs that are RANDOM
+        random_audit_course_user_group_ids = CourseCohort.objects.filter(
+            course_user_group_id__in=audit_course_user_group_ids,
+            assignment_type=CourseCohort.RANDOM
+        ).values_list('course_user_group_id', flat=True)
+
+        # Get the CourseUserGroupPartitionGroup for the above IDs, these contain the partition IDs and group IDs
+        # that are set for group_access inside of modulestore
+        random_audit_course_user_group_partition_groups = list(CourseUserGroupPartitionGroup.objects.filter(
+            course_user_group_id__in=random_audit_course_user_group_ids
+        ))
+
+        # Get the single Verified Track Cohorted Course for the course
+        verified_track = VerifiedTrackCohortedCourse.objects.get(course_key=course_key)
+
+        # If there is no verified track, raise an error
+        if not verified_track:
+            raise CommandError('No VerifiedTrackCohortedCourse found for course: %s' % course_id)
+
+        # Get the single CourseUserGroupPartitionGroup for the verified_track based on the verified_track name
+        verified_course_user_group = CourseUserGroup.objects.get(
+            course_id=course_key,
+            group_type=CourseUserGroup.COHORT,
+            name=verified_track.verified_cohort_name
+        )
+        verified_course_user_group_partition_group = CourseUserGroupPartitionGroup.objects.get(
+            course_user_group_id=verified_course_user_group.id
+        )
+
+        # Get the enrollment track CourseModes for the course
+        audit_course_mode = CourseMode.objects.get(
+            course_id=course_key,
+            mode_slug=CourseMode.AUDIT
+        )
+        verified_course_mode = CourseMode.objects.get(
+            course_id=course_key,
+            mode_slug=CourseMode.VERIFIED
+        )
+        # Verify that the enrollment track course modes exist
+        if not audit_course_mode or not verified_course_mode:
+            raise CommandError('Audit or Verified course modes are not defined for course: %s' % course_id)
+
+        for item in items:
+            # Checks whether or not an xblock is published, taken from contentstore/views/item.py create_xblock_info
+            is_library_block = isinstance(item.location, LibraryUsageLocator)
+            published = modulestore().has_published_version(item) if not is_library_block else None
```

Comment:
Since this information is only needed in a specific case, I would move doing this check until it is needed.